apiserver/facades/client/applicationoffers/applicationoffers.go

```diff
@@ -111,7 +119,35 @@ func (api *OffersAPIv5) Offer(ctx context.Context, all params.AddApplicationOffe
 		if _, err = offerBackend.ApplicationOffer(applicationOfferParams.OfferName); err == nil {
 			_, err = offerBackend.UpdateOffer(applicationOfferParams)
 		} else {
-			_, err = offerBackend.AddOffer(applicationOfferParams)
+			var addedOffer *jujucrossmodel.ApplicationOffer
+			addedOffer, err = offerBackend.AddOffer(applicationOfferParams)
+			if err != nil {
+				result[i].Error = apiservererrors.ServerError(err)
+				continue
+			}
+
+			// TODO(aflynn50): New offer permissions are handled here as there
+			// is no cross model relations service yet. These should be handled
+			// there once its created.
```

Comment:
question: Is this a jira card in the offer domain epic?

Reply:
https://warthogs.atlassian.net/browse/JUJU-6578